src/services/TestResultsService.ts

```diff
@@ -311,17 +311,45 @@ export class TestResultsService {
                   (testType.testResult === TEST_RESULT.PASS || testType.testResult === TEST_RESULT.PRS)) {
                   payload.testTypes[index] = testType;
                   if (payload.vehicleType === VEHICLE_TYPES.PSV) {
-                    if (dateFns.isEqual(mostRecentExpiryDateOnAllTestTypesBySystemNumber, new Date(1970, 1, 1))
-                        || dateFns.isBefore(mostRecentExpiryDateOnAllTestTypesBySystemNumber, dateFns.startOfDay(new Date()))
-                        || dateFns.isAfter(mostRecentExpiryDateOnAllTestTypesBySystemNumber, dateFns.addMonths(new Date(), 2))) {
-                      testType.testExpiryDate = dateFns.subDays(dateFns.addYears(new Date(), 1), 1).toISOString();
-                      payload.testTypes[index] = testType;
-                    } else if (dateFns.isToday(mostRecentExpiryDateOnAllTestTypesBySystemNumber)) {
-                      testType.testExpiryDate = dateFns.addYears(new Date(), 1).toISOString();
-                      payload.testTypes[index] = testType;
-                    } else if (dateFns.isBefore(mostRecentExpiryDateOnAllTestTypesBySystemNumber, dateFns.addMonths(new Date(), 2)) && dateFns.isAfter(mostRecentExpiryDateOnAllTestTypesBySystemNumber, new Date())) {
-                      testType.testExpiryDate = dateFns.addYears(mostRecentExpiryDateOnAllTestTypesBySystemNumber, 1).toISOString();
-                      payload.testTypes[index] = testType;
+                  // registrationAnniversary = registration date + 1 year (anniversary date as is stated in CVSB-11509)
+                  const registrationAnniversary = dateFns.addYears(payload.regnDate!!, 1);
```

Comment:
I think I should check it here also @m-ujaffer

Reply:
The bad date code is focus on Expiry Dates. This is a registration date. We can add it to make code more safe.

Reply:
I asked because I saw this change on the original bad date PR:
https://github.com/dvsa/cvs-svc-test-results/blob/86b9e1e05f752cdbbe83a0977e230d739f1fa484/src/services/TestResultsService.ts#L329